Update OP Stack genesis creation docs (#1599)
ethereum-optimism/docs: pages/operators/chain-operators/deploy/genesis.mdx

---
title: Chain artifacts creation
lang: en-US

> description: Learn how to retrieve genesis and rollup configuration files using op-deployer after L1 contract deployment.

Reviewer: ```suggestion
description: Learn how to create genesis and rollup configuration files using op-deployer after L1 contract deployment.
```

import { Callout } from 'nextra/components'

# Overview

After deploying the L1 system contracts for your OP Stack chain, you need to generate two files to run nodes on the L2 network:

*   **Genesis file** (`genesis.json`): Initializes the execution client (`op-geth`)
*   **Rollup configuration file** (`rollup.json`): Configures the consensus client (`op-node`)

This guide shows you how to generate these files from your L1 contract deployment data using `op-deployer`.

## Prerequisites

Before generating genesis and rollup configuration files, ensure you have:

1.  Deployed the L1 contracts successfully using the [`op-deployer apply`](/operators/chain-operators/tools/op-deployer#apply-deploy-your-chain) command

> 2.  Access to your deployment workspace directory (`.deployer` if you've followed the instructions in the last guide)

Reviewer 2: What's the last guide? I'd link to it as the docs in this category aren't numbered or anything

Examine the generated files to ensure they contain the expected configuration:

```bash
# View the genesis file structure
jq . .deployer/genesis.json | head -20

# View the rollup configuration
jq . .deployer/rollup.json
```

The `genesis.json` file contains the initial state of your L2 blockchain, including:

*   Chain configuration parameters
*   Initial account states
*   Genesis block properties

The `rollup.json` file contains the rollup protocol parameters, including:

*   L1 contract addresses
*   Sequencing settings
*   Protocol upgrade activation times

## Troubleshooting

If you encounter issues with the generated files:

*   [Verify](/operators/chain-operators/tools/op-deployer#verify-verify-contract-source-code-on-block-explorers) that your L1 contract deployment was successful
*   Check that you're using the correct **L2 chain ID** in the generation commands
*   Ensure your `state.json` file is up-to-date with your latest deployment

<Callout type="warning">
  Never manually edit the generated genesis or rollup files unless you fully understand the implications. Incorrect configurations can lead to consensus failures or chain security issues.
</Callout>

> ## Next Steps

Reviewer 2: ```suggestion
## Next steps
```

import { Callout } from 'nextra/components'

# Overview

<Callout type="info">

>   This guide uses `op-deployer` to generate genesis and rollup configuration files, ensuring standardization and compatibility with the Superchain.

Reviewer: I think you can remove this whole call out

</Callout>

> After deploying the L1 system contracts for your OP Stack chain, you need to generate two critical configuration files:

Reviewer: ```suggestion
After deploying the L1 system contracts for your OP Stack chain, you need to generate two files to run nodes on the L2 network:
```

*   **Genesis file** (`genesis.json`): Initializes the execution client (`op-geth`)
*   **Rollup configuration file** (`rollup.json`): Configures the consensus client (`op-node`)

This guide shows you how to generate these files from your L1 contract deployment data using `op-deployer`.

## Prerequisites

Before generating genesis and rollup configuration files, ensure you have:

> 1.  Installed `op-deployer` following the instructions in the [op-deployer documentation](/operators/chain-operators/tools/op-deployer#installation)

Reviewer: I feel like you can remove this one, but I don't feel strongly about it

2.  Deployed the L1 contracts successfully using the [`op-deployer apply`](/operators/chain-operators/tools/op-deployer#apply-deploy-your-chain) command

> 3.  Access to your deployment workspace directory (typically `.deployer`)

Reviewer: I'd recommend something like this for the context in the parentheses (`.deployer` if you've followed the instructions in the last guide)

Verify that this file exists in your workspace:

```bash
ls -la .deployer/state.json
```

### Step 2: Generate the L2 genesis file

Use the `inspect genesis` command to retrieve your L2 genesis file:

```bash
op-deployer inspect genesis --workdir .deployer <L2_CHAIN_ID> > .deployer/genesis.json
```

Replace `<L2_CHAIN_ID>` with your L2 network's chain ID.

### Step 3: Generate the rollup configuration file

Use the `inspect rollup` command to retrieve your rollup configuration file:

```bash
op-deployer inspect rollup --workdir .deployer <L2_CHAIN_ID> > .deployer/rollup.json
```

### Step 4: Verify the generated files

Examine the generated files to ensure they contain the expected configuration:

```bash
# View the genesis file structure
jq . .deployer/genesis.json | head -20

# View the rollup configuration
jq . .deployer/rollup.json
```

The `genesis.json` file contains the initial state of your L2 blockchain, including:

*   Chain configuration parameters
*   Initial account states
*   Genesis block properties

The `rollup.json` file contains the rollup protocol parameters, including:

*   L1 contract addresses

> *   Sequencing settings

Reviewer: I don't know if this is true

Author: It is @sbvegan:
**L1 contract addresses** - This is correct. The file contains addresses for contracts like:

batch_inbox_address: "0x003c9f264c1a1bd61945bf3c7464f22d3148bb83"
deposit_contract_address: "0xd2164af82dade3acf281c411775140857cb124fd"
l1_system_config_address: "0x26c5ddd3d9cdf1e6ea422a24e2c8ea6a32033b3d"
protocol_versions_address: "0x79add5713b383daa0a138d3c4780c7a1804a8090"

**Sequencing settings** - This is also correct. The file includes parameters that control the sequencing process, e.g:

block_time: 2
max_sequencer_drift: 600
seq_window_size: 3600
channel_timeout: 300

> *   Finality parameters

Reviewer: ```suggestion
*   Hardfork activations
```
